lms/djangoapps/courseware/courses.py

```diff
@@ -458,6 +466,24 @@ def get_course_assignment_due_dates(course, user, request, num_return=None, incl
     return date_blocks
 
 
+def _requires_full_access(user, block_key):
+    """
+    Returns a boolean if any child of the block_key specified has a group_access array consisting of just full_access
+    """
+    store = modulestore()
```

Comment:
Should this just take a `store` as a parameter since we technically already create one in the above method?

Reply:
Probably, was assuming it cached the store and wasn't expensive to use this way but might as well pass it in

Reply:
It definitely might, I just have no idea and better safe than sorry